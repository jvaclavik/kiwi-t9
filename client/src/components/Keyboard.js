// @flow
import React from "react"
import { View } from "react-native"
import shortid from "shortid"

// components
import { RoundedButton } from "."

// utils
import keyboardMap from "../utils/keyboardMap"
import renderStyle from "../utils/renderStyle"

// theme
import { Metrics } from "../themes"

const styles = {
  grid: {
    flex: 1,
    flexDirection: "row",
    flexWrap: "wrap",
    justifyContent: "center",
    marginHorizontal: Metrics.spacings.large,
    marginVertical: Metrics.spacings.double,
  },
  button: {
    width: "33%",
    alignItems: "center",
    justifyContent: "center",
    marginVertical: Metrics.spacings.base,
  },
}

type Props = {|
  +onNumberPress: string => void,
  style?: StyleValue,
|}

export default class Keyboard extends React.PureComponent<Props> {
  static defaultProps = {
    style: {},
  }

  render() {
    const { onNumberPress, style } = this.props
    return (
      <View style={renderStyle([styles.grid, style])}>
        {keyboardMap.map(item => (
          <View style={renderStyle(styles.button)} key={shortid.generate()}>
            <RoundedButton
              onPress={() => onNumberPress(item.number)}
              number={item.number}
              characters={item.text}
            />
          </View>
        ))}
      </View>
    )
  }
}
