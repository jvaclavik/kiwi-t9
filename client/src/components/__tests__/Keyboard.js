// @flow
import React from "react"
import renderer from "react-test-renderer"

// components
import Keyboard from "../Keyboard"

test("renders correctly with required props only", () => {
  const tree = renderer
    .create(
      <Keyboard
        onNumberPress={() => null}
        onBackspacePress={() => null}
        onResetPress={() => null}
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test("renders correctly with optional props", () => {
  const tree = renderer
    .create(
      <Keyboard
        onNumberPress={() => null}
        onBackspacePress={() => null}
        onResetPress={() => null}
        style={{ backgroundColor: "white" }}
        showBackspace
        showReset
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
