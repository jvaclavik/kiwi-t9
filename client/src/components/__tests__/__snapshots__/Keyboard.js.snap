// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with optional props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginHorizontal": 30,
        "marginVertical": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          1
        </Text>
        
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          2
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          ABC
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          3
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          DEF
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          4
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          GHI
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          5
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          JKL
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          6
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          MNO
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          7
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          PQRS
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          8
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          TUV
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          9
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          WXYZ
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#ffffff",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          RESET
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          0
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          _
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#ffffff",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#b9beca",
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          ⌫
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly with required props only 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginHorizontal": 30,
        "marginVertical": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          1
        </Text>
        
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          2
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          ABC
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          3
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          DEF
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          4
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          GHI
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          5
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          JKL
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          6
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          MNO
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          7
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          PQRS
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          8
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          TUV
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          9
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          WXYZ
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgb(24,202,167)",
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "justifyContent": "center",
            "opacity": 1,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
              "fontWeight": "200",
            }
          }
        >
          0
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          _
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 5,
          "width": "33%",
        }
      }
    />
  </View>
</View>
`;
